### PR TITLE
Cmake icestorm improvements

### PR DESCRIPTION
**Pull request by elmsfu**

### CODE REVIEW

File: ice40/CMakeLists.txt [hunk 1]
```diff
@@ -1,48 +1,87 @@
...
-  BIT_TIME ${ICETIME}
-  BIN_EXTENSION bin
-  BIT_TIME_CMD "\${BIT_TIME} -v -t -p \${INPUT_IO_FILE} -d \${DEVICE} \${OUT_BITSTREAM} -o \${OUT_TIME_VERILOG}"
-  ROUTE_CHAN_WIDTH 100
-)
+function(icestorm_setup)
+  get_target_property_required(PYTHON3 env PYTHON3)
+
+  set(ICESTORM_SRC ${symbiflow-arch-defs_SOURCE_DIR}/third_party/icestorm CACHE PATH "Path to icestorm repository")
+  set(PYUTILS_PATH ${symbiflow-arch-defs_SOURCE_DIR}/utils CACHE PATH "Path to python utils")
```
Comment (litghost) [inline ice40/CMakeLists.txt]: Not sure you want this as a CACHE var?

File: ice40/CMakeLists.txt [hunk 2]
```diff
@@ -1,48 +1,87 @@
...
+
+  get_target_property(ICEBOX_VLOG_TARGET env ICEBOX_VLOG_TARGET)
+  get_target_property(ICEPACK_TARGET env ICEPACK_TARGET)
+  get_target_property(ICETIME_TARGET env ICETIME_TARGET)
+  get_target_property(ICEBOX_HLC2ASC_TARGET env ICEBOX_HLC2ASC_TARGET)
+
+  get_target_property(ICEBOX env ICEBOX)
+  get_filename_component(ICEBOX_PATH ${ICEBOX} DIRECTORY)
+
+  set(PYPATH_ARG "PYTHONPATH=\${ICEBOX_PATH}:${PYUTILS_PATH}" CACHE STRING "environment variables for python namely PYTHONPATH")
```
Comment (litghost) [inline ice40/CMakeLists.txt]: Again, probably shoudn't be a CACHE variable.

File: ice40/CMakeLists.txt (hunk 1 above)
Comment (elmsfu) [reply ice40/CMakeLists.txt]: Done

File: ice40/CMakeLists.txt (hunk 2 above)
Comment (elmsfu) [reply ice40/CMakeLists.txt]: Done

**Comment by elmsfu** [comment]
Marking as WIP, since Travis isn't kicking off.

File: ice40/CMakeLists.txt [hunk 3]
```diff
@@ -1,48 +1,88 @@
...
+  get_target_property(PKG-CONFIG env PKG-CONFIG)
+  get_target_property(PKG-CONFIG_TARGET env PKG-CONFIG_TARGET)
+  add_thirdparty_package(
+    NAME icestorm
+    PROVIDES iceprog icebox_hlc2asc icebox_vlog icepack icetime icebox
+    BUILD_INSTALL_COMMAND "make -C ${ICESTORM_SRC} ${ICESTORM_PREFIX} PKG_CONFIG=${PKG-CONFIG} install"
+    DEPENDS ${LIBFTDI_TARGET} ${PKG-CONFIG_TARGET}
+    )
+
+  get_target_property(ICEBOX_VLOG env ICEBOX_VLOG)
```
Comment (litghost) [inline ice40/CMakeLists.txt]: Probably want get_target_property_required? get_target_property will not generate an error if unset.

**Pull request merged**